src/ray/raylet/node_manager.cc

```diff
@@ -218,20 +218,19 @@ ray::Status NodeManager::RegisterGcs() {
     RAY_CHECK_OK(status);
     // Subscribe to resource changes.
     const auto &resources_changed =
-        [this](const ClientID &id,
-               const gcs::ResourceChangeNotification &resource_notification) {
-          if (resource_notification.IsAdded()) {
+        [this](const rpc::NodeResourceChange &resource_notification) {
+          auto id = ClientID::FromBinary(resource_notification.node_id());
+          if (0 == resource_notification.deleted_resources_size()) {
             ResourceSet resource_set;
-            for (auto &entry : resource_notification.GetData()) {
+            for (auto &entry : resource_notification.updated_resources()) {
```

Comment:
Nit, can be simplified as `ResourceSet resource_set(MapFromProtobuf(resource_notification.updated_resources));

Reply:
fixed